### ISSUE TITLE
typedef of a function incorrectly output in HTML and PDF

### ISSUE THREAD

**Issue by Albrecht-S** [doxygen/doxygen#10295]
**Describe the bug**
The following C++ code
```c++
  typedef int (Int_Function2)(int, int);
```
generates the wrong output:
```c++
  typedef int() Int_Function2(int, int);
```

Note the extraneous `()` after `int`.

A function typedef w/o `()`, for instance `typedef int Int_Function2(int, int);` works as expected. This could be used as a workaround but that wouldn't solve the issue. The project I'm working on has lots of typedef's using the former syntax.

**Expected behavior**
Output should be
```c++
  typedef int Int_Function2(int, int);
```

**Screenshots**
Please see PDF file with more comments:
[Function_Typedef.pdf](https://github.com/doxygen/doxygen/files/12561967/Function_Typedef.pdf)

**To Reproduce**
Self contained example:
[typedef.tar.gz](https://github.com/doxygen/doxygen/files/12561976/typedef.tar.gz)

**Version**
This issue is visible in doxygen 1.10.0, self-built from git master, commit a3cba2660e0464c4a584377ccd80f5830add7277, but also in earlier releases. I don't think it's a regression in recent releases (but I may be wrong with this).
Platform: Linux, Debian 12 (Bookworm), included system doxygen version is 1.9.4.
Tested also with doxygen 1.8.14 (same effect).

**Additional context**
Related issues: #10136, #9944, #7060. All these seem to indicate problems parsing typedefs, particularly function typedefs with redundant `()` as in my example. I went back until about July 2018 to find similar issues (found these three mentioned issues).

The example posted above uses by default the smallest code to show the issue with (a) correct and (b) failing typedef. Both syntax variants are IMHO legit.

I found even more issues when I used different typedef's, some from the cppref site (see included example code). You can enable more examples by setting the macro `MORE_EXAMPLES` to `1` as indicated in the included source file `test1.cxx`. One of the examples from cppref uses a comma-separated list of typedef's that shows even more - only loosely related - issues (incorrect type propagation from one element of the list to later elements). I wanted to mention these here since I found these issues and they seem to be related.

Thanks for this great tool and I appreciate the work all devs are putting into it.

**Comment by Albrecht-S** [comment]
@albert-github Thanks for reading and for classifying it as a bug.

**Comment by doxygen** [comment]
This is also listed as known problem [here](https://www.doxygen.nl/manual/trouble.html) (see bullet starting with "Redundant braces can confuse doxygen in some cases"). 

Probably in this particular case something is possible, but it is a very hard problem in the general case without proper type analysis which doxygen does not do (and often cannot do in case the input is incomplete).

**Comment by doxygen** [comment]
@Albrecht-S Please verify if the referenced commit fixes the issues you found. Do not close the issue, this will be done automatically when the next official version is released.

**Comment by Albrecht-S** [comment]
@doxygen Yep, current master (tested: c29b0e69b907ec1ae2bdee01498fdb46cb8a6989) works as expected. From my POV this issue is fixed. Suggestion: please check if the related issues in my original post are fixed as well and close if appropriate.

Thank you very, very much for the quick fix.

Current master also fixes the following `list of typedef's` which had issues in 1.9.8:
```c++
/** more complicated typedef's. */
typedef int int_t, *intp_t, (&fp)(int, long), arr1_t[10];
```

Since the fix is in `master`, is it correct to assume that it will be in 1.10.0 which is the current version info of `master` ?

**Comment by albert-github** [comment]
> Since the fix is in master, is it correct to assume that it will be in 1.10.0 which is the current version info of master ?

This is correct

**Comment by albert-github** [comment]
Good suggestion to check the related issues:
- #9944 this looks to me as solved
- #10136 I think this is still wron9
- #7060 I think this is still wrong.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
